sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -389,97 +390,77 @@ export class ManagementClient extends LinkEntity {
+            "[%s] Acquiring lock to get the management req res link.",
+            this._context.connectionId
+          );
 
-          const remainingOperationTimeoutInMs = retryTimeoutInMs - timeTakenByInit;
-
-          const sendRequestOptions: SendRequestOptions = {
-            abortSignal: options.abortSignal,
-            requestName: options.requestName,
-            timeoutInMs: remainingOperationTimeoutInMs
-          };
-
-          count++;
-          if (count !== 1) {
-            // Generate a new message_id every time after the first attempt
-            request.message_id = generate_uuid();
-          } else if (!request.message_id) {
-            // Set the message_id in the first attempt only if it is not set
-            request.message_id = generate_uuid();
-          }
+          const initOperationStartTime = Date.now();
 
           try {
-            const result = await this._mgmtReqResLink!.sendRequest(request, sendRequestOptions);
-            resolve(result);
+            await waitForTimeoutOrAbortOrResolve({
+              actionFn: () => {
+                return defaultLock.acquire(this.managementLock, () => {
+                  return this._init();
+                });
+              },
+              abortSignal: options?.abortSignal,
+              timeoutMs: retryTimeoutInMs,
+              timeoutMessage: `The request with message_id "${request.message_id}" timed out. Please try again later.`
+            });
           } catch (err) {
             const translatedError = translate(err);
             logger.warning(
-              "[%s] An error occurred during send on management request-response link with address '%s': %s",
+              "[%s] An error occurred while creating the management link %s: %s",
               this._context.connectionId,
-              this.address,
+              this.name,
               `${translatedError?.name}: ${translatedError?.message}`
             );
             logErrorStackTrace(translatedError);
-            reject(translatedError);
+            throw translatedError;
           }
-        });
+          timeTakenByInit = Date.now() - initOperationStartTime;
+        }
+
+        const remainingOperationTimeoutInMs = retryTimeoutInMs - timeTakenByInit;
+
+        const sendRequestOptions: SendRequestOptions = {
+          abortSignal: options.abortSignal,
+          requestName: options.requestName,
+          timeoutInMs: remainingOperationTimeoutInMs
+        };
+
+        count++;
+        if (count !== 1) {
+          // Generate a new message_id every time after the first attempt
+          request.message_id = generate_uuid();
+        } else if (!request.message_id) {
+          // Set the message_id in the first attempt only if it is not set
+          request.message_id = generate_uuid();
+        }
+
+        try {
+          const result = await this._mgmtReqResLink!.sendRequest(request, sendRequestOptions);
+          return result;
```

Comment:
Since all we're doing is returning the result, and we don't have a `finally`, could we also ditch the `catch` and reduce the whole `try/catch` to this?
```suggestion
          return this._mgmtReqResLink!.sendRequest(request, sendRequestOptions);
```

Reply:
Fair point. Done in https://github.com/Azure/azure-sdk-for-js/pull/13826/commits/ec7f720c474b9973d4f14ebe6388b5fda98dd680